.github/workflows/integration.yml

```diff
@@ -0,0 +1,233 @@
+name: 'integration'
+
+on:
+  push:
+    branches:
+    - 'main'
+  pull_request:
+    branches:
+    - 'main'
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
```

Comment:
Wondering if deploys are in progress and a new commit is pushed, would it still run cleanup?

Reply:
That's an interesting question... from what I can tell, it cancels the job even if `always()` is used.

Reply:
Maybe as a follow up we could probably do with a cron workflow that deletes *all* functions where create time > 30m?